backend-rust/src/bin/ccdscan-api.rs

```diff
@@ -155,26 +214,39 @@ async fn main() -> anyhow::Result<()> {
             }
             cancel_token.cancel();
         }
+        result = &mut node_collector_task => {
+            error!("Node collector task stopped.");
+            if let Err(err) = result? {
+                error!("Node collector error: {}", err);
+            }
+            cancel_token.cancel();
+        }
     }
     info!("Shutting down");
     // Ensure all tasks have stopped
-    let _ = tokio::join!(monitoring_task, queries_task, pgnotify_listener);
+    let _ = tokio::join!(monitoring_task, queries_task, pgnotify_listener, node_collector_task);
     Ok(())
 }
 
+#[derive(Clone)]
+struct HealthState {
+    pool:                 sqlx::PgPool,
+    node_status_receiver: Receiver<Option<Vec<NodeStatus>>>,
+}
+
 /// GET Handler for route `/health`.
 /// Verifying the API service state is as expected.
 async fn health(
-    axum::extract::State(pool): axum::extract::State<sqlx::PgPool>,
+    axum::extract::State(state): axum::extract::State<HealthState>,
 ) -> axum::Json<serde_json::Value> {
-    match migrations::ensure_compatible_schema_version(&pool, SUPPORTED_SCHEMA_VERSION).await {
-        Ok(_) => axum::extract::Json(json!({
-            "status": "ok",
-            "database": "connected"
-        })),
-        Err(err) => axum::extract::Json(json!({
-            "status": "error",
-            "database": format!("not connected: {}", err)
-        })),
-    }
+    let node_status_connected = state.node_status_receiver.borrow().is_some();
+    let database_connected =
+        migrations::ensure_compatible_schema_version(&state.pool, SUPPORTED_SCHEMA_VERSION)
```

Comment:
The error message here contains information of why it is failing, so maybe it would be useful to keep it.
But it depends a bit on how the health endpoint is used, I guess.

Reply:
It is basically solely for the kubernetes control plane. It is not even externally exposed.

So relevant errors should be logged.